### ISSUE TITLE
Display of the planned start time, end time and remaining time.

### ISSUE THREAD

**Issue by iqtreiber** [cpvalente/ontime#514]
During the planning and implementation of events, the question arises more and more often whether/how/when other event-parts (speaker slot) would still be free.

Here a start time and end time would be very helpful, which automatically displays the remaining unplanned time or calculates the theoretical end time, if it differs from the planned end time.

In many cases an event and related invitations take place at fixed times. Example from 10AM to 5PM. During this event planning and implementation it would be helpful to display the remaining free time or even too much used time.

Example:
START TIME: 10AM
END TIME: 5PM

Info from the Ontime App during event planning, >>.
1) Time hh:mm left (free) until End time (5PM)
2) your end time would be earlier as scheduled (e.g. 4PM)
3) your end time would have overtime as scheduled (e.g. 6PM)

**Comment by cpvalente** [comment]
Hi @iqtreiber, this is definitely something that I would personally like to improve. But it has proven to be a design challenge, both in terms of UI and code architecture.

I feel this to be complex with a lot of moving parts and there will be clarifications needed.
From my experience with other software, this implies that the slot allocated to the rundown (10AM-5PM) in your example is continuous, ie: an event starts right after another finishes.

How can we deal with parts of the rundown being unscheduled?
How would this play with the current (manual) delay flow? I feel stroingly that a good decision here would allow users to have manual control of delays in addition to the automated delay system

We have some attempts at discussing this in #100,  a long-standing ticket that I would like to prioritise.

**Comment by iqtreiber** [comment]
Yes, I can understand that this is not a small issue and has a lot of dependencies.
Even I don't know what way is the best to describe a possible solution for this requirement.

But maybe "somehow" the event cue's can be aligned to horizontal timeline from left to right.
Of course the complex starts with the issues you are asking
1) cues unscheduled? >>status label = can not be aligned to a timeline
2) manual delays? >> should be displayed aligned to the timeline

<img width="1409" alt="Phase 2" src="https://github.com/cpvalente/ontime/assets/9744544/1c641f8d-ca4a-42e6-af61-ef181d81639d">

**Comment by cpvalente** [comment]
Hi @iqtreiber, I have made some tests with the idea of having a timeline in the /editor route.

The feature goals where defined as 

- Provide visual feedback as to the the relative duration of the events
- Provide visual feedback as to the progress of the runtime
- Highlight runtime delay (difference between clock and runtime) along with its origin

This is a rough test, UI would definitely be improved and would also include delays and better visualisation
<img width="1504" alt="Screenshot 2023-11-29 at 20 45 58" src="https://github.com/cpvalente/ontime/assets/34649812/dc69219f-69c4-48bb-a443-71c442b7a79b">


> - Provide visual feedback as to the the relative duration of the events

With anything more than around 15 elements, the timeline was unreadable and the relationship became meaningless. The solution I found was to force the timeline to overflow off screen and scroll (like the elements in the /operator view). 
At this point, it no longer serves the outline purpose since we cannot use it to have an overview of the entire rundown

> - Provide visual feedback as to the progress of the runtime

Following up on the point above, we could ensure that the timeline provides feedback over the next x amount of events. 
I fount this to have questionable value since Ontime has very different usages. For some users the next 5 events can count up to 5 minutes, while others 5 events could be half a day of programming
With showing only a few events I also found that it didnt add much to the existing rundown list

- Highlight runtime delay (difference between clock and runtime) along with its origin

I still think this is a good point, but is invalidated by the fact that we wouldnt be able to show the entire rundown

From my investigations it seems that our interface is unsuitable for this feature.
Altogether, I think this is a good feature, but doesnt rank much higher than a "nice to have". We have limited work capacity and there are likely other features which would add more time

At his point my inclination is to consider off scope for the app., but I am hoping that you can share some thoughts. 


Also @hrueger, I see that you had a timeline in your fork, maybe you have some opinions

**Comment by iqtreiber** [comment]
Hi @cpvalente,

many thanks for your professional research and investigation of possible feasibilities.
I understand very well the results and impacts and of course agree that this way is not a solution.

But, please let me rewrite this request by changing the target.
As we do not follow any kind of scale and proportion scaling, would be possible to show
1) the difference between a event line in +/- mm:ss
2) in case mm:ss is the current difference - allow to fill with a new event-line, break-line
3) show the difference, color-high-lighted, between each event-line
4) in case a over-lap (takes longer), show the difference

Keeping my question from original with small changes
1) is it possible to show at the main timer, the absolute end-time
2) show at the main timer, the absolute start-time

many thanks
BR Stefan

**Comment by hrueger** [comment]
Just a quick note here:
In my opinion (this might be completely personal, though) a timeline-like overview would be great. If we look at the VSCode Minimap for example, you can also not read any code there, but it helps with general overview.
I feel like we could have a very small bar (like 1 - 2 rem high) at the top without text (maybe title and start / end times in a popover on hover) which just displays boxes with the event colors. If we add a red play head for example, you could get a pretty good overview of how far you're into the event, if there are one or two more breaks, etc
Does that make sense?
Other than that, I agree with all the points here.

**Comment by cpvalente** [comment]
Hi @iqtreiber , could you please provide some feedback as to how the changes in v3 satisfy this specification?

**Comment by iqtreiber** [comment]
Yes, this new changes are fulfil very well the request. many thanks!
Issue can be closed.
